### ISSUE TITLE
[BUG-REPORT] Vaex.close() does not close file handles.

### ISSUE THREAD

**Issue by Hasham04** [vaexio/vaex#1939]
**Description**
df.close() does not close file handles.

```
from shutil import rmtree 
df=vaex.open('s3_path', fs_options{})#multiple parquet files.
df.close()
rmtree('/home/user/myname/.vaex',ignore_errors=False) #this says that one of the files directories in the folder is not empty.
rmtree should delete even if directories are not empty...
```

This has something to do with .open because when I run this command at the start of the pipeline it deletes the folder.
```
try:
        rmtree('/home/user/myname/.vaex',ignore_errors=False) # this deletes the folder if it exists
except:
        print('no dir')
df=vaex.open('s3_path', fs_options{})
df.close()

```
I have tried

```
for d in df.dataset:
  d.close() # error str object has no attribut close.
also tried

# error df has no attribute dfs.
for d in df.dfs:
  d.close() 
```

**Software information**
 - Vaex version 4.8:
 - installed via pip
 - windows 10:

**Additional information**
Please state any supplementary information or provide additional context for the problem (e.g. screenshots, data, etc..).

**Comment by Hasham04** [comment]
What happens if when I 

rmtree('/home/user/myname/.vaex',ignore_errors=True) # True instead of false 

all the files inside the folders get deleted but the folders remain. The .vaex folder in the home directory is now 0 bytes.

```
from shutil import rmtree 
rmtree('/home/user/myname/.vaex',ignore_errors=True) # this will delete the folder .vaex from the home directory??
df=vaex.open('s3_path', fs_options{})#multiple parquet files.
df.close()
rmtree('/home/user/myname/.vaex',ignore_errors=True) #this deletes all files inside the folder but not the folders themselves
```

**Comment by JovanVeljanoski** [comment]
There is no bug here. Let me explain briefly:

Then you open a file from a remote source (like S3 for example), the portions of that file that you are using a streamed to your local machine, and stored in `/home/user/username/.vaex/`. If you access the same file again, vaex knows which parts of the original file in the cloud have been stored, and access will be as fast as local access (because it is). 
This is all documented btw - you should read through the vaex documentation first before opening issues. 

That folder (i.e. `/home/user/username/.vaex/`) is used for various other things, and vaex will always create it while it is in use. You can manually or programmatically delete the files caches if they are no longer needed, in any way you want. 

Now, vaex memory maps files like hdf5, arrow, parquet. Meaning those files are not read and kept in memory but you are reading them from disk as needed. So if you want to sever the connection between vaex and some file on disk, so that you can safely delete it, move it, or do any other operation unrelated to vaex, you need vaex to close the handle (or connection) to the file. This is achieved with `df.close()`. _This does not delete the file on disk itself, vaex will never do that._ I don't know why you think it should.  Same goes for the `/home/user/username/.vaex/` folder. 

I do not use `rmtree` (does this come from `shutil`??) so i can comment on that. Maybe ask things on stackoverflow for general python help. 

So in summary:
  - `df.close()` does indeed close the file handle, but the file itself (or the file-cache) is not removed from disk, and that is how it should be.
  - the `/home/user/username/.vaex/` folder is needed for vaex to work, so if it does not exist it will be created. You may delete the file-cache inside if it is no longer needed manually, programmatically or in whatever way you want.
  - Please read the documentation before opening issues

Also, there is already a thread in the discussion section on this topic opened by you. - please do not make duplicate threads.

**Comment by Hasham04** [comment]
Hey Jovan, firstly I would like to apologize for creating the duplicate request. I thought it might it was an issue so I thought it should be in the issues section. I will keep this in mind. But thank you for explaining. I am very new to using vaex but I appreciate your patience. In fact, this was the first issue I ever created on Github. Thank you for your help as always :).
